codegen/test/rtc/src/tmp_dir.cpp

```diff
@@ -31,18 +35,56 @@ std::string unique_string(const std::string& prefix)
 }
 
 tmp_dir::tmp_dir(const std::string& prefix)
-    : path(std::filesystem::temp_directory_path() /
-           unique_string(prefix.empty() ? "ck-rtc" : "ck-rtc-" + prefix))
+    : path(get_tmp_dir_path() + "/" + unique_string(prefix.empty() ? "ck-rtc" : "ck-rtc-" + prefix))
 {
-    std::filesystem::create_directories(this->path);
+    new_dir(this->path);
 }
 
+std::string tmp_dir::get_tmp_dir_path()
+{
+    // use getenv to get the path of the tmp dir
+    const char* tmp_dir_path = std::getenv("TMPDIR");
+
+    if(tmp_dir_path == nullptr)
+    {
+        return "/tmp";
+    }
+    return tmp_dir_path;
+}
+
+// TODO: finish this method
+void tmp_dir::new_dir(std::string dir_path)
+{
+    int created = mkdir(dir_path.c_str(), 0755);
```

Comment:
If we want to create a temporary file, we can use `std::filesystem::temp_directory_path()`, which calls `mkdir`. Alternatively, we can use `mkdtemp` from POSIX:
```cpp
std::string pattern = "tmep/nnn-xxxx";
if (mkdtemp(pattern.data()) == NULL) { }
```